pulsar-common/src/main/java/org/apache/pulsar/common/compression/AirliftUtils.java

```diff
@@ -25,10 +25,10 @@
  */
 public abstract class AirliftUtils {
 
-    static ByteBuffer ensureAirliftSupported(ByteBuffer encodedNio, int uncompressedLength) {
+    static ByteBuffer ensureAirliftSupported(ByteBuffer encodedNio) {
         if (!encodedNio.isDirect() && !encodedNio.hasArray()) {
             // airlift needs a raw ByteArray
-            ByteBuffer copy = ByteBuffer.allocate(uncompressedLength);
+            ByteBuffer copy = ByteBuffer.allocate(encodedNio.capacity());
```

Comment:
I am not sure I understand the fix here.
'uncompressedLength' may be usually bigger than the size of the buffer passed to this method, because data is compressed.

What about using the 'max' between the two values?

Reply:
Yes, using max is better.

Reply:
Currently, if the source text is no repeated, the compressed data length is original data length + 1, I haven't found out the reason, but I think the capacity is enough for the ByteBuffer copy operation.